realm/src/androidTest/java/io/realm/DynamicRealmObjectTest.java

```diff
@@ -66,6 +65,26 @@ protected void tearDown() throws Exception {
         BOOLEAN, SHORT, INT, LONG, BYTE, FLOAT, DOUBLE, STRING, BINARY, DATE, OBJECT, LIST;
     }
 
+    public void testIllegalInputObjectThrows() {
+        try {
+            new DynamicRealmObject(null);
+        } catch (IllegalArgumentException ignored) {
```

Comment:
You are missing `fail()` here and below. Otherwise this will pass even if an exception isn't thrown.